Source/CarthageKit/Xcode.swift

```diff
@@ -355,6 +355,39 @@ public enum SDK: String {
 	}
 }
 
+public func < (lhs: SDK, rhs: SDK) -> Bool {
+	switch (lhs, rhs) {
+	// Prefer Mac OS X SDK.
+	case (.MacOSX, _):
+		return true
+
+	case (_, .MacOSX):
+		return false
+
+	// Prefer device SDK over simulator SDK in same platform.
+	case (.iPhoneOS, .iPhoneSimulator):
+		return true
+
+	case (.iPhoneSimulator, .iPhoneOS):
+		return false
+
+	case (.watchOS, .watchSimulator):
+		return true
+
+	case (.watchSimulator, .watchOS):
+		return false
+
+	case (.tvOS, .tvSimulator):
+		return true
+
+	case (.tvSimulator, .tvOS):
+		return false
+
+	case _:
+		return true
```

Comment:
I don't love returning true in this case. I'd prefer to establish a total order by specifying the order of different platforms. We could either sort by `self.platform.rawValue` here or compare the `rawValue` like you suggested above. If we do the latter, I think  it'd be good to add a comment explaining that simulators need to come last.